### PR TITLE
[BUGFIX] Kafka producer failing with DatadogMiddleware

### PR DESCRIPTION
**Pull request by lmagalhaes**
#### Description

There is an issue (more like an easter egg) preventing Kafka message from being produced when `DatadogMiddleware` is activated.

The issue happens basically because `DatadogMiddleware` expectes payload to be a dictionary so it can inject the Datadog traces in the message. However, Kafka is sending a Message.value attribute, which is a string causing the following error.

```
Cannot create property '_meta' on string '*' 
```

This PR makes sure that both Kafka producers and Kafra consumers sends the payload to all middlewares as a dictionary, instead of a dictionary.

----
#### Changes
_List the main changes in this PR._

----

#### Testing Instructions
_Outline the steps to test or reproduce the PR._

----

#### Screenshots
_(if appropriate, they help the reviewer visualise the changes)_

----

### CODE REVIEW

**Review by johndagostino — DISMISSED** [review]

File: packages/steveo/test/consumer/kafka_test.ts [hunk 1]
```diff
@@ -358,4 +358,60 @@ describe('runner/kafka', () => {
...
+        middleware: [
+          {
+            publish: () => {},
+            consume: (context) => {
+              receivedContext = context.payload;
+            },
+          },
+        ]
+      },
+      // @ts-ignore
```
Comment (sajidanower23) [inline packages/steveo/test/consumer/kafka_test.ts:393]: This `ts-ignore` is not necessary 😄

File: packages/datadog/src/index.ts [hunk 2]
```diff
@@ -59,6 +50,20 @@ export class DataDogMiddleware implements Middleware {
...
+    let tags = {};
+    if (typeof context.payload !== 'string') {
+      tags = { ...context.payload };
+    }
+
+    return {
+      resource: context.topic,
+      tags,
+    };
+  }
```
Comment (sajidanower23) [inline packages/datadog/src/index.ts:66]: Nitpick, but you can simplify this function like so:
```suggestion
  private createBaseTraceOptionsFromContext(
    context: MiddlewareContext
  ): tracer.TraceOptions & SpanOptions {
    return {
      resource: context.topic,
      tags: typeof context.payload !== "string" ? { ...context.payload } : {},
    };
  }
```

File: packages/steveo/test/consumer/kafka_test.ts [hunk 3]
```diff
@@ -358,4 +358,60 @@ describe('runner/kafka', () => {
...
+        publish: () => {},
+        subscribe: subscribeStub,
+      }),
+      emit: sandbox.stub(),
+      events: {
+        emit: sandbox.stub(),
+      },
+    };
+    const messagePayload: Record<string, string> = {
+      my: 'payload'
```
Comment (sajidanower23) [inline packages/steveo/test/consumer/kafka_test.ts:375]: Let's also add a test for when the payload is a string. That is the edge case that this PR is aiming to fix, yeah?

File: packages/datadog/src/index.ts (hunk 2 above)
Comment (treefitty) [reply packages/datadog/src/index.ts:66]: i agree on the nitpick, but would add a little more meaning to the test value:

```
 private createBaseTraceOptionsFromContext(
	    context: MiddlewareContext
	  ): tracer.TraceOptions & SpanOptions {
  	  const validPayload = typeof context.payload !== "string";
  	  const tags = validPayload ? { ...context.payload } : {};
  	  
	    return {
	      resource: context.topic,
	      tags,
	    };
	  }
```

this makes debugging and logging so much easier than mashing everything into one line, and provides context as to what the condition is actually intending to test

File: packages/steveo/test/consumer/kafka_test.ts (hunk 3 above)
Comment (lmagalhaes) [reply packages/steveo/test/consumer/kafka_test.ts:375]: Only for the producer. The consumer will always receive a Json parseable payload.

File: packages/datadog/src/index.ts (hunk 2 above)
Comment (lmagalhaes) [reply packages/datadog/src/index.ts]: Done

**Review by treefitty — APPROVED** [review]
As sajid and I mentioned, probably worth tidying up that code block though

**Review by sugendran — APPROVED** [review]

**Review by sajidanower23 — APPROVED** [review]
